google/cloud/aiplatform/utils/gcs_utils.py

```diff
@@ -163,3 +163,52 @@ def stage_local_data_in_gcs(
     )
 
     return staged_data_uri
+
+
+def create_gcs_directory_for_pipeline_artifacts(
```

Comment:
this is great

Reply:
Thank you!

P.S. I've decided to split this into a two-stage approach. It's better to not perform any changes in constructor or factory method. Now the bucket creation is delayed until the user calls `job.submit()`.